《花开伊吕波剧场版：甜蜜的家》无法被正确匹配并同步

```log
[2026/04/21 20:12:45.437] [DEBUG] 接收到Emby同步请求：{'Title': 'DiskStation 上 jacksen168 已标记 花开伊吕波剧场版：甜蜜的家 为已播放', 'Date': '2026-04-21T12:12:45.3839645Z', 'Event': 'item.markplayed', 'Severity': 'Info', 'User': {'Name': 'jacksen168', 'Id': '1ac2781c41034c35885e6c29c28b8cd5'}, 'Item': {'Name': '花开伊吕波剧场版：甜蜜的家', 'OriginalTitle': '劇場版 花咲くいろは HOME SWEET HOME', 'ServerId': '88d3e2edef944171a6a549b81efb8241', 'Id': '5741', 'DateCreated': '2025-02-07T16:05:22.0000000Z', 'DateModified': '2025-02-07T16:05:22.0000000Z', 'Container': 'mkv', 'SortName': '花开伊吕波剧场版：甜蜜的家', 'PremiereDate': '2013-03-08T16:00:00.0000000Z', 'ExternalUrls': [{'Name': 'IMDb', 'Url': 'https://www.imdb.com/title/tt2404237'}, {'Name': 'TheMovieDb', 'Url': 'https://www.themoviedb.org/movie/229756'}, {'Name': 'TheTVDB', 'Url': 'https://thetvdb.com/dereferrer/movie/50949'}, {'Name': 'Trakt', 'Url': 'https://trakt.tv/search/tmdb/229756?id_type=movie'}], 'Path': '/volume2/video/Movies/花开伊吕波剧场版：甜蜜的家/Hanasaku Iroha Home Sweet Home.mkv', 'OfficialRating': 'PG-13', 'Overview': '时间缓缓流淌，16岁的东京少女松前绪花在外婆经营的温泉旅馆喜翠庄度过了一段快乐而难忘的时光。石川县迎来秋高气爽的时节，和仓结名被福屋老板娘拜托来到喜翠屋修行，这个快乐外向的女孩令这里增添不少的生气。结名在打扫时不小心弄乱存放杂物的房间，绪花赶来收拾时，偶然发现了一本豆丁爷爷的日记。令她颇感意外的是，里面记录了母亲皋月年轻时候的叛逆经历，以及她在十数年前与外婆争吵、和绪花的父亲邂逅的往事。', 'Taglines': [], 'Genres': ['动画', '剧情', '喜剧'], 'CommunityRating': 7.2, 'RunTimeTicks': 40170150000, 'Size': 3481319244, 'FileName': 'Hanasaku Iroha Home Sweet Home.mkv', 'Bitrate': 6933146, 'ProductionYear': 2013, 'RemoteTrailers': [{'Url': 'https://www.youtube.com/watch?v=MnhP4I_CAr4'}], 'ProviderIds': {'Imdb': 'tt2404237', 'Tvdb': '50949', 'Tmdb': '229756', 'official website': 'http://www.hanasakuiroha.jp/'}, 'IsFolder': False, 'ParentId': '4003', 'Type': 'Movie', 'Studios': [{'Name': 'P.A.WORKS', 'Id': 4108}, {'Name': 'Bandai Visual', 'Id': 4109}, {'Name': 'Lantis', 'Id': 4110}], 'GenreItems': [{'Name': '动画', 'Id': 1946}, {'Name': '剧情', 'Id': 1949}, {'Name': '喜剧', 'Id': 1947}], 'TagItems': [], 'UserData': {'PlaybackPositionTicks': 0, 'PlayCount': 1, 'IsFavorite': False, 'LastPlayedDate': '2026-04-21T12:12:45.0000000Z', 'Played': True}, 'PrimaryImageAspectRatio': 0.6666666666666666, 'ImageTags': {'Primary': 'bd0bcb13e53f0c96157aa87ea6da082e', 'Logo': 'a3303c6dcc5650588ed4e64e2242054b', 'Thumb': '7576df4216388c2b7d03d48cc5d84ed5'}, 'BackdropImageTags': ['e4fc139cf307c65e9eac7a334a501847'], 'MediaType': 'Video', 'Width': 1920, 'Height': 1080}, 'Server': {'Name': 'DiskStation', 'Id': '88d3e2edef944171a6a549b81efb8241', 'Version': '4.9.3.0'}}
[2026/04/21 20:12:45.437] [DEBUG] Emby重新组装JSON报文：media_type='movie' title='花开伊吕波剧场版：甜蜜的家' ori_title=None season=1 episode=1 release_date='2013-03-08' user_name='jacksen168' source='emby'
[2026/04/21 20:12:45.438] [INFO] 接收到同步请求：media_type='movie' title='花开伊吕波剧场版：甜蜜的家' ori_title=None season=1 episode=1 release_date='2013-03-08' user_name='jacksen168' source='emby'
[2026/04/21 20:12:45.439] [DEBUG] 使用缓存的映射配置，共 2 个映射
[2026/04/21 20:12:45.439] [DEBUG] 正在查找番剧 ID: title='花开伊吕波剧场版：甜蜜的家', ori_title=None, release_date='2013-03-08', season=1
[2026/04/21 20:12:45.439] [DEBUG] 开始尝试完全匹配...
[2026/04/21 20:12:45.440] [DEBUG] 使用内存缓存数据 (命中次数: 13)
[2026/04/21 20:12:45.437] [INFO] Emby同步任务 emby_12_1776773565 已提交到异步队列
[2026/04/21 20:12:45.458] [INFO] Emby webhook已提交异步任务: emby_12_1776773565
[2026/04/21 20:12:46.912] [DEBUG] 处理了 8480 个项目，找到 0 个完全匹配，1 个部分匹配
[2026/04/21 20:12:46.912] [DEBUG] 没有找到完全匹配的番剧，尝试进行模糊匹配...
[2026/04/21 20:12:46.912] [DEBUG] 找到 1 个可能的匹配项:
[2026/04/21 20:12:46.912] [DEBUG]   1. 花咲くいろは, 中文翻译: 花开伊吕波, 花开物语, 花开色叶, 匹配度: 0.5043859649122806
[2026/04/21 20:12:46.912] [DEBUG] 未找到匹配的番剧 ID
[2026/04/21 20:12:46.913] [WARNING] SSL证书验证已禁用，这会降低安全性。建议仅在代理环境下出现SSL错误时使用。
[2026/04/21 20:12:46.913] [DEBUG] BangumiApi 初始化 - 代理参数: 无, SSL验证: False
[2026/04/21 20:12:52.754] [ERROR] bgm: 未查询到番剧信息，跳过
bgm: item.title='花开伊吕波剧场版：甜蜜的家' item.ori_title=None premiere_date='2013-03-08'
```

看了眼BGM 名为[《剧场版 花开伊吕波 HOME SWEET HOME》](https://bgm.tv/subject/49892)
TMDB: [《花开伊吕波剧场版：甜蜜的家》](https://www.themoviedb.org/movie/229756-home-sweet-home)

属于我也不知道要怎么修才好的问题了,目前只能使用手动映射才能正常打格子。先报告先了。

--- a/app/utils/bangumi_data.py
+++ b/app/utils/bangumi_data.py
@@ -327,8 +327,7 @@ class BangumiData:
 
             # 快速预筛选：默认要求有简中翻译；若提供了 ori_title，仍可对无 zh-Hans 的条目做日文原标题匹配
             missing_zh = (
-                "titleTranslate" not in item
-                or "zh-Hans" not in item["titleTranslate"]
+                "titleTranslate" not in item or "zh-Hans" not in item["titleTranslate"]
             )
             if title and missing_zh:
                 ori_ok = ori_title and str(ori_title).strip() and item.get("title")

--- a/app/utils/bangumi_data.py
+++ b/app/utils/bangumi_data.py
@@ -325,11 +325,15 @@ class BangumiData:
         for item in self._parse_data():
             processed_count += 1
 
-            # 快速预筛选：检查是否有中文翻译
-            if title and (
-                "titleTranslate" not in item or "zh-Hans" not in item["titleTranslate"]
-            ):
-                continue
+            # 快速预筛选：默认要求有简中翻译；若提供了 ori_title，仍可对无 zh-Hans 的条目做日文原标题匹配
+            missing_zh = (
+                "titleTranslate" not in item
+                or "zh-Hans" not in item["titleTranslate"]
+            )
+            if title and missing_zh:
+                ori_ok = ori_title and str(ori_title).strip() and item.get("title")
+                if not ori_ok:
+                    continue
 
             # 一次性计算所有相似度，避免重复计算
             match_info = self._calculate_match_info(

--- a/app/utils/data_util.py
+++ b/app/utils/data_util.py
@@ -93,10 +93,11 @@ def extract_emby_data(emby_data):
         else:
             logger.debug("未找到PremiereDate字段，将尝试从bangumi-data获取日期信息")
         title = (item.get("Name") or "").strip()
+        ori = item.get("OriginalTitle")
         return CustomItem(
             media_type="movie",
             title=title,
-            ori_title=None,
+            ori_title=ori if ori and str(ori).strip() else None,
             season=1,
             episode=1,
             release_date=release_date,

--- a/tests/utils/test_bangumi_data_comprehensive.py
+++ b/tests/utils/test_bangumi_data_comprehensive.py
@@ -144,6 +144,30 @@ class TestBangumiDataMatching:
         # 断言：由于防误判锁生效拒绝了日期择优，可信度必须退回到 False，交给后端去爬树
         assert result[2] is False
 
+    @patch("app.utils.bangumi_data.BangumiData._preload_data_to_memory")
+    @patch("app.utils.bangumi_data.BangumiData._parse_data")
+    def test_find_bangumi_id_ori_title_without_zh_hans(
+        self, mock_parse_data, mock_preload
+    ):
+        """条目无简中翻译时，仍可用 ori_title 与日文 title 精确匹配（剧场版等）"""
+        data = BangumiData()
+        mock_parse_data.return_value = [
+            {
+                "title": "劇場版サンプル映画",
+                "begin": "2020-03-15",
+                "sites": [{"site": "bangumi", "id": "123456"}],
+            },
+        ]
+        result = data.find_bangumi_id(
+            title="某中文片名",
+            ori_title="劇場版サンプル映画",
+            release_date="2020-03-15",
+            season=1,
+        )
+        assert result is not None
+        assert result[0] == "123456"
+        assert result[1] == "劇場版サンプル映画"
+
 
 class TestBangumiDataTitle:
     """标题处理测试"""

--- a/tests/utils/test_data_util.py
+++ b/tests/utils/test_data_util.py
@@ -170,11 +170,33 @@ class TestExtractEmbyData:
 
         assert result.media_type == "episode"
         assert result.title == "测试番剧"
+        assert result.ori_title is None
         assert result.season == 2
         assert result.episode == 10
         assert result.release_date == "2024-01-15"
         assert result.user_name == "test_user"
         assert result.source == "emby"
+
+    def test_extract_emby_data_episode_does_not_use_original_title(self):
+        """Emby 剧集 OriginalTitle 为分集名，不作为 ori_title（避免条目匹配跑偏）"""
+        from app.utils.data_util import extract_emby_data
+
+        emby_data = {
+            "Event": "item.markplayed",
+            "User": {"Name": "test_user"},
+            "Item": {
+                "Type": "Episode",
+                "SeriesName": "某动画",
+                "SeriesOriginalTitle": "シリーズ原文",
+                "OriginalTitle": "第3話 サブタイトル",
+                "ParentIndexNumber": 1,
+                "IndexNumber": 3,
+                "PremiereDate": "2024-02-01T00:00:00.0000000Z",
+            },
+        }
+        result = extract_emby_data(emby_data)
+        assert result.title == "某动画"
+        assert result.ori_title is None
 
     def test_extract_emby_data_no_premiere_date(self):
         """测试无发行日期"""
@@ -195,6 +217,7 @@ class TestExtractEmbyData:
         result = extract_emby_data(emby_data)
 
         assert result.release_date == ""
+        assert result.ori_title is None
 
     def test_extract_emby_data_movie(self):
         from app.utils.data_util import extract_emby_data
@@ -211,9 +234,47 @@ class TestExtractEmbyData:
         result = extract_emby_data(emby_data)
         assert result.media_type == "movie"
         assert result.title == "剧场版 Y"
+        assert result.ori_title is None
         assert result.season == 1
         assert result.episode == 1
         assert result.release_date == "2024-07-01"
+
+    def test_extract_emby_data_movie_original_title(self):
+        """Emby 电影应携带 OriginalTitle 作为 ori_title（剧场版等日文检索）"""
+        from app.utils.data_util import extract_emby_data
+
+        emby_data = {
+            "Event": "item.markplayed",
+            "User": {"Name": "test_user"},
+            "Item": {
+                "Type": "Movie",
+                "Name": "花开伊吕波剧场版：甜蜜的家",
+                "OriginalTitle": "劇場版 花咲くいろは HOME SWEET HOME",
+                "PremiereDate": "2013-03-08T16:00:00.0000000Z",
+            },
+        }
+        result = extract_emby_data(emby_data)
+        assert result.media_type == "movie"
+        assert result.title == "花开伊吕波剧场版：甜蜜的家"
+        assert result.ori_title == "劇場版 花咲くいろは HOME SWEET HOME"
+        assert result.release_date == "2013-03-08"
+
+    def test_extract_emby_data_movie_original_title_blank(self):
+        """OriginalTitle 为空或仅空白时 ori_title 为 None"""
+        from app.utils.data_util import extract_emby_data
+
+        emby_data = {
+            "Event": "item.markplayed",
+            "User": {"Name": "test_user"},
+            "Item": {
+                "Type": "Movie",
+                "Name": "剧场版 X",
+                "OriginalTitle": "   ",
+                "PremiereDate": "2024-01-01T00:00:00.0000000Z",
+            },
+        }
+        result = extract_emby_data(emby_data)
+        assert result.ori_title is None
 
     def test_extract_emby_data_movie_no_premiere_date(self):
         """电影无 PremiereDate 时 release_date 为空"""
@@ -228,6 +289,7 @@ class TestExtractEmbyData:
         assert result.media_type == "movie"
         assert result.title == "剧场版 Z"
         assert result.release_date == ""
+        assert result.ori_title is None
 
 
 class TestExtractJellyfinData:

--- a/tests/utils/test_data_util.py
+++ b/tests/utils/test_data_util.py
@@ -170,7 +170,7 @@ class TestExtractEmbyData:
 
         assert result.media_type == "episode"
         assert result.title == "测试番剧"
-        assert result.ori_title is None
+        assert result.ori_title == " "
         assert result.season == 2
         assert result.episode == 10
         assert result.release_date == "2024-01-15"
@@ -178,7 +178,7 @@ class TestExtractEmbyData:
         assert result.source == "emby"
 
     def test_extract_emby_data_episode_does_not_use_original_title(self):
-        """Emby 剧集 OriginalTitle 为分集名，不作为 ori_title（避免条目匹配跑偏）"""
+        """剧集仍用单空格占位；即使有 OriginalTitle 也不写入（其为分集名）"""
         from app.utils.data_util import extract_emby_data
 
         emby_data = {
@@ -196,7 +196,7 @@ class TestExtractEmbyData:
         }
         result = extract_emby_data(emby_data)
         assert result.title == "某动画"
-        assert result.ori_title is None
+        assert result.ori_title == " "
 
     def test_extract_emby_data_no_premiere_date(self):
         """测试无发行日期"""
@@ -217,7 +217,7 @@ class TestExtractEmbyData:
         result = extract_emby_data(emby_data)
 
         assert result.release_date == ""
-        assert result.ori_title is None
+        assert result.ori_title == " "
 
     def test_extract_emby_data_movie(self):
         from app.utils.data_util import extract_emby_data

